test/parallel/test-repl-programmatic-history-setup-history.js

```diff
@@ -0,0 +1,272 @@
+
+// File paths
+const historyFixturePath = fixtures.path('.node_repl_history');
+const historyPath = tmpdir.resolve('.fixture_copy_repl_history');
+const historyPathFail = fixtures.path('nonexistent_folder', 'filename');
+const defaultHistoryPath = tmpdir.resolve('.node_repl_history');
+const emptyHiddenHistoryPath = fixtures.path('.empty-hidden-repl-history-file');
+const devNullHistoryPath = tmpdir.resolve('.dev-null-repl-history-file');
+// Common message bits
+const prompt = '> ';
+const replDisabled = '\nPersistent history support disabled. Set the ' +
+                     'NODE_REPL_HISTORY environment\nvariable to a valid, ' +
+                     'user-writable path to enable.\n';
+const homedirErr = '\nError: Could not get the home directory.\n' +
+                   'REPL session history will not be persisted.\n';
+const replFailedRead = '\nError: Could not open history file.\n' +
+                       'REPL session history will not be persisted.\n';
+
+const tests = [
+  {
+    env: { NODE_REPL_HISTORY: '' },
+    test: [UP],
+    expected: [prompt, replDisabled, prompt]
+  },
+  {
+    env: { NODE_REPL_HISTORY: ' ' },
+    test: [UP],
+    expected: [prompt, replDisabled, prompt]
+  },
+  {
+    env: { NODE_REPL_HISTORY: historyPath },
+    test: [UP, CLEAR],
+    expected: [prompt, `${prompt}'you look fabulous today'`, prompt]
+  },
+  {
+    env: {},
+    test: [UP, '21', ENTER, "'42'", ENTER],
+    expected: [
+      prompt,
+      '2', '1', '21\n', prompt,
+      "'", '4', '2', "'", "'42'\n", prompt,
+    ],
+    clean: false
+  },
+  { // Requires the above test case
+    env: {},
+    test: [UP, UP, UP, DOWN, ENTER],
+    expected: [
+      prompt,
+      `${prompt}'42'`,
+      `${prompt}21`,
+      prompt,
+      `${prompt}21`,
+      '21\n',
+      prompt,
+    ]
+  },
+  {
+    env: { NODE_REPL_HISTORY: historyPath,
+           NODE_REPL_HISTORY_SIZE: 1 },
+    test: [UP, UP, DOWN, CLEAR],
+    expected: [
+      prompt,
+      `${prompt}'you look fabulous today'`,
+      prompt,
+      `${prompt}'you look fabulous today'`,
+      prompt,
+    ]
+  },
+  {
+    env: { NODE_REPL_HISTORY: historyPathFail,
+           NODE_REPL_HISTORY_SIZE: 1 },
+    test: [UP],
+    expected: [prompt, replFailedRead, prompt, replDisabled, prompt]
+  },
+  {
+    before: function before() {
+      if (common.isWindows) {
+        const execSync = require('child_process').execSync;
+        execSync(`ATTRIB +H "${emptyHiddenHistoryPath}"`, (err) => {
+          assert.ifError(err);
+        });
+      }
+    },
+    env: { NODE_REPL_HISTORY: emptyHiddenHistoryPath },
+    test: [UP],
+    expected: [prompt]
+  },
+  {
+    before: function before() {
+      if (!common.isWindows)
+        fs.symlinkSync('/dev/null', devNullHistoryPath);
+    },
+    env: { NODE_REPL_HISTORY: devNullHistoryPath },
+    test: [UP],
+    expected: [prompt]
+  },
+  { // Make sure this is always the last test, since we change os.homedir()
```

Comment:
:sweat:

Reply:
Yeah, 99% of the content of this file is just a copy of other 2 test files: `test-repl-programmatic-history.js` and `test-repl-persistent-history.js`.   
Removing the comment from this one because in here it is not necessary, it looks like